apps/nextjs-example/components/WalletButtons.tsx

```diff
@@ -3,26 +3,44 @@ import {
   WalletReadyState,
   Wallet,
   isRedirectable,
+  WalletName,
 } from "@aptos-labs/wallet-adapter-react";
+import { Dispatch, SetStateAction } from "react";
 
-const WalletButtons = () => {
+interface WalletButtonsProps {
+  setErrorAlertMessage: Dispatch<SetStateAction<string>>;
+}
+
+const WalletButtons = (props: WalletButtonsProps) => {
   const { wallets } = useWallet();
 
   return (
     <>
       {wallets.map((wallet: Wallet) => {
-        return walletView(wallet);
+        return walletView(wallet, props.setErrorAlertMessage);
       })}
     </>
   );
 };
 
-const walletView = (wallet: Wallet) => {
+const walletView = (
+  wallet: Wallet,
+  setErrorAlertMessage: Dispatch<SetStateAction<string>>
+) => {
   const { connect } = useWallet();
   const isWalletReady =
     wallet.readyState === WalletReadyState.Installed ||
     wallet.readyState === WalletReadyState.Loadable;
   const mobileSupport = wallet.deeplinkProvider;
+
+  const onWalletConnectRequest = async (walletName: WalletName) => {
+    try {
+      await connect(walletName);
+    } catch (error: any) {
+      setErrorAlertMessage(error);
+    }
+  };
```

Comment:
If there is a way to differentiate between "already connected" and another case, I'd probably make "already connected" a warning alert rather than an error alert.  Just for a UX perspective